proto/encode.go

```diff
@@ -139,6 +144,9 @@ func (p *Buffer) EncodeFixed64(x uint64) error {
 // This is the format for the
 // fixed32, sfixed32, and float protocol buffer types.
 func (p *Buffer) EncodeFixed32(x uint64) error {
+	if x > math.MaxUint32 {
```

Comment:
https://play.golang.org/p/xSeqrvM8Bye 😟 I think the naïve use of this function will cause this change to break a lot of code if taken as is.